app/models/concerns/foreman_teamdynamix/host_extensions.rb

```diff
@@ -0,0 +1,22 @@
+module ForemanTeamdynamix
+  module HostExtensions
+    extend ActiveSupport::Concern
+
```

Comment:
We should move most of this code to orchestration and follow the pattern used by compute/dns etc.
I was thinking there has to ba a more standard way to tap into the progress bar etc when creating a host.
This is how the compute and dns etc are handled in core foreman: https://github.com/theforeman/foreman/tree/006929d1057d5b59458d4fbf99cd20853c198ffd/app/models/concerns/orchestration

Reply:
The puppetdb module was recently updated to add this too: https://github.com/theforeman/puppetdb_foreman/blob/master/app/models/concerns/orchestration/puppetdb.rb

Reply:
to be done with issue#18 PR